[MIG] medical_center: Migrate to 10.0

kellylougheed opened the pull request:
Depends:
- [x] #119

lasley committed:
[MIG] medical_center: Migrate to v10
* Upgrade medical_center to v10

jcdrubay commented on medical_center/models/medical_center.py:
```diff
@@ -0,0 +1,32 @@
+        )
+        with open(img_path, 'r') as image:
+            base64_image = image.read().encode('base64')
+            return tools.image_resize_image_big(base64_image)
```
`return` should be outside the `with` because we are done with the `image` already.

jcdrubay commented on medical/models/medical_abstract_entity.py:
```diff
@@ -0,0 +1,207 @@
+                )
+                with open(img_path, 'r') as image:
+                    base64_image = image.read().encode('base64')
+                    return odoo.tools.image_resize_image_big(base64_image)
```
@lasley The source of the problem ;)

`return` should be outside the `with` because we are done with the `image` already.

jcdrubay reviewed (commented):
LGTM, only a tiny adjustment required. most important is probably to take it into account in medical.

lasley commented:
Rebased now that depend has been merged

lasley commented on medical_center/models/medical_center.py:
```diff
@@ -0,0 +1,32 @@
+
+    @api.model
+    def _get_default_image(self, vals):
+        res = super(MedicalCenter, self)._get_default_image(vals)
```
This is missing tests. Take a look at the patient one - https://github.com/OCA/vertical-medical/blob/10.0/medical/tests/test_medical_patient.py#L120

jcdrubay commented on medical_center/__manifest__.py:
```diff
@@ -0,0 +1,27 @@
+    "data": [
+        "views/medical_center.xml",
+        "views/medical_patient.xml",
+        "views/medical_menu.xml",
```
It does not follow the contribution guidlines: https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#file-naming


But I wonder if contribution guidelines should not be updated... something more like: 

- addons/<my_module_name>/
  - menu
    - `For a menu: <root_menu_name>_menu.xml`

jcdrubay replied:
Let me know what you think and I will create a PR to CONTRIBUTING.md if you agree.

lasley replied:
There's actually a debate about this one right now, so we're in somewhat of a state of limbo. I'm on the side of adding the suffix though, which is inline with current docs. I recommend you state your feelings on the subject in OCA/maintainer-tools#191 😄 

For now @kellylougheed please make these files follow the guidelines, adding a `_view`, `_demo`, or `_data` suffix where applicable please and thanks!

jcdrubay replied:
Done: https://github.com/OCA/maintainer-tools/pull/191#issuecomment-273362738

lasley replied:
@kellylougheed - As PSC lead of this repo, I have overridden the contribution guidelines for file naming that @jcdrubay pointed to. Please switch back to the dedicated menu file.

While the argument can be made that maybe this doesn't make sense for other repos, it most certainly does here. Medical gets deep, and these menus are incredibly difficult to manage. We need the isolation.

lasley replied:
Oh whoops, I see the rename that was made. Disregard me and my lack of attention, adding _view is good!

rgarnau commented:
There were 2 bugs already fixed related to the fields 'supplier' (PR https://github.com/OCA/vertical-medical/pull/178) and 'is_center' (PR https://github.com/LasLabs/vertical-medical/pull/216).

rgarnau commented:
@lasley Travis is failing because you need to rebase the branch of this PR with OCA/10.0. For example, we detected that method test_get_default_image_encoded exists  in OCA/10.0 but not in this branch.

rgarnau commented:
@lasley @kellylougheed Disregard last comment. We will send a PR to your branch with the needful fixes.

Commit:
[10.0][medical_center] fix use of default_image (#219)

* This commit fixes an error that occurs when trying to select a parent
company to a medical center. The field is_center is not used anymore.
It does not make sense to restrict the selection of parent company only
to partners that are not medical centers, because the parent company
may just be a larger medical center.

* fix use of default_image_encoded method

rgarnau reviewed (approved):
Technical and functional review.

JordiBForgeFlow commented:
@kellylougheed @lasley This is ready to merge.
